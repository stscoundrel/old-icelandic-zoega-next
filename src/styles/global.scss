@import 'reset-css';
@import 'variables';
@import 'media-queries';
@import 'typography';
@import 'containers';
@import 'buttons';

body,
html {
  max-width: 100%;
  overflow-x: hidden;
  scroll-behavior: smooth;
}

html {
  box-sizing: border-box;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}