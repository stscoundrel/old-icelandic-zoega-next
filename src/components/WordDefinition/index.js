import { lettersToRunes } from 'younger-futhark'
import { capitalize } from 'lib/utils/strings'
import { addAbbreviationsToContent } from 'lib/services/abbreviations'
import Abbreviations from 'components/Abbreviations'
import Crosslinks from 'components/Crosslinks'
import styles from './WordDefinition.module.scss'

export default function WordDefinition({ data, abbreviations, crosslinks }) {
  const { word, definitions } = data

  return (
    <article className={styles.section}>
      <header>
        <h1 lang="non">{capitalize(word)}</h1>

        <small className={styles.subHeading}>
          Old Icelandic Dictionary - {word.toLowerCase()}
        </small>
        <p>Meaning of Old Icelandic word <em>&quot;{word}&quot;</em> in English.</p>
      </header>

      <p>As defined by <em>A Concise Dictionary of Old Icelandic</em> (Geir ZoeÌˆga):</p>

      {definitions.length > 1 && <p><dfn className="capitalize">{word}</dfn> Old Icelandic word can mean:</p>}
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong></dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={{
              __html: addAbbreviationsToContent(definition, abbreviations),
            } }
          ></dd>
        </dl>
      ))}

      <p>Possible runic inscription in <em>Younger Futhark:</em>
          <span className={styles.rune}>{ lettersToRunes(word) }</span>
        </p>

      <Abbreviations abbreviations={abbreviations} />
      <Crosslinks crosslinks={crosslinks} />
    </article>
  )
}
