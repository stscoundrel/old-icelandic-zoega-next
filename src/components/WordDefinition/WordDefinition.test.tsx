import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordDefinition from './index'
import styles from './WordDefinition.module.scss'

const word = {
  word: 'leynidyrr',
  definitions: [
    'f. pl. <i>secret door</i>.',
  ],
  slug: 'leynidyrr',
}

const abbreviations = [
  {
    abbreviation: 'f.',
    explanation: 'feminine.',
  },
  {
    abbreviation: 'pl.',
    explanation: 'plural.',
  },
]

const crosslinks = [
  {
    url: 'https://cleasby-vigfusson-dictionary.vercel.app/word/abbadis',
    source: 'old-norse',
  },
  {
    url: 'https://old-norwegian-dictionary.vercel.app/word/abbadis',
    source: 'old-norwegian',
  },
]
const runes = 'ᛚᛁᚢᚾᛁᛏᚢᚱᚱ'

describe('WordDefinition component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        crosslinks={crosslinks}
        runes={runes}
      />,
    )
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        crosslinks={crosslinks}
        runes={runes}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        crosslinks={crosslinks}
        runes={runes}
      />,
    )
    const { root } = tree

    expect(root.findByType('h1').children).toEqual(['Leynidyrr'])
  })

  test('Has correct amount of definitions', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        crosslinks={crosslinks}
        runes={runes}
      />,
    )
    const { root } = tree

    expect(root.findAllByProps({ className: styles.abbreviation }).length).toEqual(2)
  })
})
