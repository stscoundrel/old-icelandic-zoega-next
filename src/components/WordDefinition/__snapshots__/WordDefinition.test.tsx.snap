// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordDefinition component Matches snapshot 1`] = `
<article
  className="section"
>
  <header>
    <h1
      lang="non"
    >
      Leynidyrr
    </h1>
    <small
      className="subHeading"
    >
      Old Icelandic Dictionary - 
      leynidyrr
    </small>
    <p>
      Meaning of Old Icelandic word 
      <em>
        "
        leynidyrr
        "
      </em>
       in English.
    </p>
  </header>
  <p>
    As defined by 
    <em>
      A Concise Dictionary of Old Icelandic
    </em>
     (Geir Zoëga):
  </p>
  <dl
    className="definitionList"
  >
    <dt>
      <strong>
        leynidyrr
      </strong>
    </dt>
    <dd
      className="itemDescription"
      dangerouslySetInnerHTML={
        {
          "__html": "<abbr title="feminine.">f.</abbr> <abbr title="plural.">pl.</abbr> <i>secret door</i>.",
        }
      }
    />
  </dl>
  <p>
    Possible runic inscription in 
    <em>
      Younger Futhark:
    </em>
    <span
      className="rune"
    >
      ᛚᛁᚢᚾᛁᛏᚢᚱᚱ
    </span>
  </p>
  <div
    className="abbreviations"
  >
    <div
      className="column"
    >
      <h4>
        Abbreviations used:
      </h4>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            f.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          feminine.
        </dd>
      </dl>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            pl.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          plural.
        </dd>
      </dl>
    </div>
  </div>
  <h4>
    Also available in related dictionaries:
  </h4>
  <p>
    This headword also appears in dictionaries of other languages closely related to Old Icelandic.
  </p>
  <ul
    className="list"
  >
    <li
      className="listItem"
    >
      <a
        href="https://cleasby-vigfusson-dictionary.vercel.app/word/abbadis"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Norse - Cleasby & Vigfusson Dictionary
      </a>
    </li>
    <li
      className="listItem"
    >
      <a
        href="https://old-norwegian-dictionary.vercel.app/word/abbadis"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Norwegian - Johan Fritzner's Dictionary
      </a>
    </li>
  </ul>
</article>
`;
